File: snippets/fsharp/System/Convert/ToString/tostring5.fs
module tostring5

// <Snippet26>
open System

type Temperature(temperature: decimal) =
    member _.Celsius =
        temperature

    member _.Kelvin =
        temperature + 273.15m

    member _.Fahrenheit =
        Math.Round(temperature * 9m / 5m + 32m |> decimal, 2)

    override _.ToString() =
        temperature.ToString("N2") + " °C"

let cold = Temperature -40
let freezing = Temperature 0
let boiling = Temperature 100

printfn $"{Convert.ToString(cold, null)}"
printfn $"{Convert.ToString(freezing, null)}"
printfn $"{Convert.ToString(boiling, null)}"
// The example dosplays the following output:
//       -40.00 °C
//       0.00 °C
//       100.00 °C
// </Snippet26>

File: snippets/csharp/System/Convert/ToString/tostring5.cs
// <Snippet26>
using System;

public class Temperature
{
    private decimal m_Temp;

    public Temperature(decimal temperature) => this.m_Temp = temperature;

    public decimal Celsius => this.m_Temp;

    public decimal Kelvin => this.m_Temp + 273.15m;

    public decimal Fahrenheit => Math.Round((decimal)(this.m_Temp * 9 / 5 + 32), 2);

    public override string ToString() => m_Temp.ToString("N2") + " °C";
}

public class Example
{
    public static void Main()
    {
        Temperature cold = new(-40);
        Temperature freezing = new(0);
        Temperature boiling = new(100);

        Console.WriteLine(Convert.ToString(cold, null));
        Console.WriteLine(Convert.ToString(freezing, null));
        Console.WriteLine(Convert.ToString(boiling, null));
    }
}
// The example dosplays the following output:
//       -40.00 °C
//       0.00 °C
//       100.00 °C
// </Snippet26>
